flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/dataSet/DataSetWindowAggregateRule.scala

```diff
@@ -49,7 +49,13 @@ class DataSetWindowAggregateRule
       throw TableException("GROUPING SETS are currently not supported.")
     }
 
-    !distinctAggs && !groupSets && !agg.indicator
```

Comment:
This check must not be removed. Instead we should return with `false` from `matches`